sbi/inference/posteriors/likelihood_based_posterior.py

```diff
@@ -408,7 +479,9 @@ def np_potential(self, theta: np.array) -> ScalarFloat:
         theta = torch.as_tensor(theta, dtype=torch.float32)
 
         # Notice opposite sign to pyro potential.
-        return self.log_likelihood(theta).cpu() + self.prior.log_prob(theta)
+        return self.log_likelihood(
+            theta, track_gradients=track_gradients
+        ).cpu() + self.prior.log_prob(theta)
 
     def pyro_potential(
```

Comment:
I named it `posterior_potential` because it is used by `np_slice` and by `rejection`. It is also quite general: it takes in a tensor `theta` and evaluates the `posterior` log-prob (thus the name). Contrary to that, `pyro_potential` is quite tailored to `pyro` (negative log-prob as potential, dealing with the `dict` that every pyro variable has to be in...)

Reply:
yeah, right. lgtm then